### PR TITLE
Bhr 35 - mocked and unstyled

### PR DESCRIPTION
**Pull request by RobertHebel**

### CODE REVIEW

File: frontend/src/actions/bookActions.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { LOAD_BOOKS_SUCCESS } from '../constants/actionTypes';
+import { getAllBooks } from '../../mockAPI/books';
+
+export const loadBooksSuccess = books => ({
```
Comment (adam-golab) [inline frontend/src/actions/bookActions.js]: Try to create asynchronous action with replaced function `getAllBooks` to Promise based function

File: frontend/src/containers/BooksTable.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import BookListRow from './BooksListRow';
+import PropTypes from 'prop-types';
+
+class BooksTable extends React.Component {
+  static propTypes = {
+    books: PropTypes.arrayOf(
+      PropTypes.shape( {
```
Comment (adam-golab) [inline frontend/src/containers/BooksTable.js]: Unnecessary space before `{`.

File: frontend/src/containers/BooksTable.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        ISBN: PropTypes.number.isRequired,
+        title: PropTypes.string.isRequired,
+        author: PropTypes.string.isRequired,
+        copies: PropTypes.number.isRequired,
+        cover: PropTypes.string.isRequired,
+      }, ),
```
Comment (adam-golab) [inline frontend/src/containers/BooksTable.js]: Unnecessary space after `},`

File: frontend/src/containers/BooksTable.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    ),
+  };
+
+  render() {
+    const books = this.props.books;
+    return (
+      <table className="table">
```
Comment (adam-golab) [inline frontend/src/containers/BooksTable.js]: `className="books-table"`

File: frontend/src/containers/BooksTable.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+        </tbody>
+      </table>
+
+    );
+  }
+}
+
+function mapStateToProps(state) {
+  return {
+    books: Object.assign([], state.books),
```
Comment (adam-golab) [inline frontend/src/containers/BooksTable.js]: What is `Object.assign` here for? Why not just `{ books: state.books }`

File: frontend/src/index.js [hunk 6]
```diff
@@ -10,12 +10,15 @@ import configureStore from './store/configureStore';
...
 import './styles/styles.scss';
 
+import { loadBooks } from './actions/bookActions';
 
 const store = configureStore();
 
 // Create an enhanced history that syncs navigation events with the store
 const history = syncHistoryWithStore(browserHistory, store);
 
+store.dispatch(loadBooks());
```
Comment (adam-golab) [inline frontend/src/index.js]: It mustn't be here. When we run `index.js` first time, a user is not authenticated. So there should be no possibility to send requests to API. You should dispatch this action in `componentWillMount` method in `BooksTable` or even better in `BooksPage` (component that are connected in router)

File: frontend/src/routes.js [hunk 7]
```diff
@@ -3,13 +3,15 @@ import { Route, IndexRoute } from 'react-router';
 
 import App from 'containers/App';
 import Login from 'containers/Login';
+import BooksTable from 'containers/BooksTable';
```
Comment (adam-golab) [inline frontend/src/routes.js]: I suggest creating new component only for page, something like `BooksPage`

File: frontend/src/reducers/booksReducer.js [hunk 8]
```diff
@@ -0,0 +1,10 @@
+import * as types from '../constants/actionTypes';
```
Comment (adam-golab) [inline frontend/src/reducers/booksReducer.js]: Use explicit imports. With current import eslint won't help with finding errors

File: frontend/src/containers/BooksTable.js (hunk 3 above)
Comment (RobertHebel) [inline frontend/src/containers/BooksTable.js]: what about comma-spacing eslint rule?

File: frontend/src/containers/BooksTable.js (hunk 2 above)
Comment (RobertHebel) [inline frontend/src/containers/BooksTable.js]: fixed

File: frontend/src/containers/BooksTable.js (hunk 4 above)
Comment (RobertHebel) [inline frontend/src/containers/BooksTable.js]: changed

File: frontend/src/reducers/booksReducer.js (hunk 8 above)
Comment (RobertHebel) [inline frontend/src/reducers/booksReducer.js]: changed

File: frontend/src/containers/BooksTable.js (hunk 3 above)
Comment (RobertHebel) [inline frontend/src/containers/BooksTable.js]: I moved `),` to new line

File: frontend/src/actions/bookActions.js [hunk 9]
```diff
@@ -0,0 +1,9 @@
+import { LOAD_BOOKS, LOAD_BOOKS_SUCCESS, LOAD_BOOKS_ERROR } from '../constants/actionTypes';
+import { getAllBooks } from '../../mockAPI/books';
+
+export function loadBooks() {
+  return {
+    types: [LOAD_BOOKS, LOAD_BOOKS_SUCCESS, LOAD_BOOKS_ERROR],
+    promise: books => getAllBooks(),
```
Comment (adam-golab) [inline frontend/src/actions/bookActions.js]: Here won't be `books` in param. `promise` is a function called in `reduxPromise` middleware, precisely in line `promise({ getState, dispatch, client: apiClient })`. So the available params are `getState`, `dispatch` and `client`, or you can get all of them with one non-destructuring assignment. 
Here you don't need `client` or any other param from middleware, so it should be a function without params. 
```
promise: () => getAllBooks(),
```

File: frontend/src/containers/Books.js [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+    this.renderBooksList = this.renderBooksList.bind(this);
+  }
+
+  renderBooksList(books) {
+    return (<ul>
+      {books.map(book =>
+          this.renderRow(book)
+        )}
+    </ul>
+    );
```
Comment (adam-golab) [inline frontend/src/containers/Books.js]: Try to be consistent. If you open brackets together with `<ul>` then you should close it together as well.
Look for lines: 44, 54, 63 and 71. There is so much inconsistency. Could you fix it?

File: frontend/src/reducers/booksReducer.js [hunk 11]
```diff
@@ -0,0 +1,41 @@
+import { LOAD_BOOKS, LOAD_BOOKS_SUCCESS, LOAD_BOOKS_ERROR } from '../constants/actionTypes';
+
+export default function(state=[], action) {
```
Comment (adam-golab) [inline frontend/src/reducers/booksReducer.js]: The initial state mustn't be an array if later you change it to an object. And remember fields that are connected in components should be always present in the state. So for example if you have a component that have `connect(state => {books: state.books.books})(YourComponent)` then field `state.books.books` should have some value from the very beginning. So in the initial state should be something like `{books: []}` or other value with the same type as destination value (`null` for objects, `[]` for arrays, `false` for boolean)

File: frontend/src/reducers/booksReducer.js [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+  switch (action.type) {
+    case LOAD_BOOKS:
+      return {
+        ...state,
+        isLoading: {
+          ...state.isLoading,
+          loadingBooks: true,
```
Comment (adam-golab) [inline frontend/src/reducers/booksReducer.js]: There is no need to duplicate `loading`. Field `isLoading.loadingBooks` looks odd. Naming it `isLoading.books` is enough. The same with errors ;)

File: frontend/test/reducers/booksReducer.test.js [hunk 13]
```diff
@@ -0,0 +1,61 @@
...
+    title: 'Kotlin for Android Developers: Learn Kotlin',
+    author: 'Antonio Leiva',
+    copies: 2,
+    cover: 'https://8.allegroimg.com/original/011d1d/8877ea844bdd955979ecb2809f48',
+  },
+];
+
+describe('Books Reducer', () => {
+  it('loading books should be equal true when passed LOAD_BOOKS action', () => {
+    const state = [];
```
Comment (adam-golab) [inline frontend/test/reducers/booksReducer.test.js]: As above, the initial state should be an object with all fields.

File: frontend/test/reducers/booksReducer.test.js [hunk 14]
```diff
@@ -0,0 +1,61 @@
+import { LOAD_BOOKS, LOAD_BOOKS_SUCCESS, LOAD_BOOKS_ERROR } from '../../src/constants/actionTypes';
+import booksReducer from '../../src/reducers/booksReducer';
+import { expect } from 'chai';
+
```
Comment (adam-golab) [inline frontend/test/reducers/booksReducer.test.js]: Could you add a test case for the initial state? I mean testing if after initialization state have correct shape and fields. Something like:
```
expect(booksReducer(undefined, { type: '@@INIT' })).to.be.deep.equal(initialState.books);
```

File: frontend/src/containers/Books.js [hunk 15]
```diff
@@ -0,0 +1,86 @@
...
+          <div className="books-list__row__details">
+            <p><b>{book.title}</b></p>
+            <p>{book.author}</p>
+          </div>
+        </div>
+      </li>
+    );
+  }
+
+  render() {
```
Comment (caderek) [inline frontend/src/containers/Books.js]: It's not a part of the task to make a nice view for it, we can keep it right now, but you have to remember - we do only things that are planned in the sprint, in other cases we could be late with all the tasks. The reason why we decided to make it simple JSON <pre> prompt is that we will probably use library for tables on which Mariusz is working on.

**Pull request merged**